lusSTR/wrappers/filter.py

```diff
@@ -412,30 +433,19 @@ def get_at(df, locus):
     return at
 
 
-def main(
-    input,
-    output_type,
+def process_input(
+    input_name,
+    outpath,
     profile_type,
     data_type,
-    output_dir,
-    info,
-    separate,
+    output_type,
     nofilters,
-    strand,
+    separate,
     custom,
+    sex,
+    info,
 ):
```

Comment:
I've never had this happen before- @standage when I specify ```nofilters``` as ```True``` in the config, it's passed as ```True``` to the function but then is changed to ```False```, so the ```no filters``` if statement is bypassed. Why is that happening?

Reply:
I do notice that you name the parameter `filters` in the Snakemake rule params block, but `nofilters` in the config file and the Python code. That's a possible source of confusion, but it doesn't explain this behavior. And otherwise I can't see any obvious reason the variable's value would be changed.

Reply:
if I do ```print(nofilters)``` in main() it's True, if I print it in process_input() it's False. 🫤

Reply:
I had changed the order of the function arguments..... 🤦‍♀️

Reply:
Ah, that'll do it. For functions with that many arguments, I almost always try to find which arguments have a reasonable default value and convert them to keyword arguments, which reduces that kind of issue.

Reply:
I added some keyword arguments.